### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by flyingfishflash** [mooltiverse/nyx#196]
Hi having an issue I can't quite figure out.

I was trying to have a setup where both nyx via gradle and nyx via binary could be executed against the same configuration.

However I've found my .nyx.yaml configuration works when picked up by gradle, but not when executing 'nyx' from the CLI.

It looks to be related to scanning the git commit history.

Notably, the error only occurs when the _mainline_ releaseType is selected.

nyx configuration:
```yaml
---
dryRun: true
preset: "simple"
verbosity: "DEBUG"
releasePrefix: "v"
commitMessageConventions:
  enabled:
    - conventionalCommitsCustom
  items:
    conventionalCommitsCustom:
      expression: "(?m)^(?<type>[a-zA-Z0-9_]+)(!)?(\\((?<scope>[a-z ]+)\\))?:( (?<title>.+))$(?s).*"
      bumpExpressions:
        major: "(?s)(?m)^[a-zA-Z0-9_]+(!|.*^(BREAKING( |-)CHANGE: )).*"
        minor: "(?s)(?m)^feat(?!!|.*^(BREAKING( |-)CHANGE: )).*"
        patch: "(?s)(?m)(^fix|^refactor)(?!!|.*^(BREAKING( |-)CHANGE: )).*"
releaseTypes:
  enabled:
    - mainline
    - internal
  items:
      mainline:
        filterTags: "^({{configuration.releasePrefix}})?([0-9]\\d*)\\.([0-9]\\d*)\\.([0-9]\\d*)$"
        gitPush: "true"
        gitTag: "true"
        matchBranches: "^(master|main)$"
        matchEnvironmentVariables:
          CI: "^true$"                 # only match this release type when running on CI servers
        matchWorkspaceStatus: "CLEAN"  # only match this release type when the repository is clean
        publish: "true"
      internal:
        collapseVersions: true
        collapsedVersionQualifier: "{{#lower}}{{branch}}{{/lower}}"
        filterTags: "^({{configuration.releasePrefix}})?([0-9]\\d*)\\.([0-9]\\d*)\\.([0-9]\\d*)$"
        gitCommit: "false"
        gitPush: "false"
        gitTag: "false"
        publish: "false"
```

nyc CLI debug output:
```
DEBU[0000] selecting the command to run                 
DEBU[0000] no command has been passed on the command line, running the default INFER 
DEBU[0000] Nyx.run(INFER)                               
DEBU[0000] Nyx.infer()                                  
DEBU[0000] running command 'INFER'                      
DEBU[0000] looking up command instance 'INFER' from cache 
DEBU[0000] no command instance 'INFER' found in cache   
DEBU[0000] creating new command instance 'INFER'        
DEBU[0000] instantiating the Git repository in '/home/gregory/repositories/ledger' 
DEBU[0000] instantiating the initial state              
DEBU[0000] new Infer command object                     
DEBU[0000] checking whether the Infer command is up to date 
DEBU[0000] the Infer command is not up to date because the internal state has no version yet or the state version doesn't match the version previously generated by Infer 
DEBU[0000] command 'INFER' is not up to date, running... 
DEBU[0000] running the Infer command...                 
DEBU[0000] clearing the state from Infer outputs        
DEBU[0000] resolving the release type among enabled ones 
DEBU[0000] evaluating release type: 'mainline'          
DEBU[0000] release type 'mainline' specifies a match branches template '^(master|main)$' that evaluates to regular expression: '^(master|main)$' 
DEBU[0000] current branch 'main' successfully matched by release type 'mainline' matchBranches regular expression '^(master|main)$' 
DEBU[0000] evaluating environment variable 'CI' as required by release type 'mainline' 
DEBU[0000] environment variable 'CI' value successfully matched by release type 'mainline' regular expression '^true$' 
DEBU[0000] checking repository clean status             
DEBU[0002] repository clean status is: 'true' ('')      
DEBU[0002] current repository status 'CLEAN' successfully matched by release type 'mainline' matchWorkspaceStatus filter 'CLEAN' 
DEBU[0002] release type 'mainline' has been selected    
DEBU[0002] current Git branch is 'main'                 
DEBU[0002] walking the commit history...                
DEBU[0002] walking commit history. Start commit boundary is 'not defined'. End commit boundary is 'not defined' 
DEBU[0002] upon merge commits only the first parent is considered. 
DEBU[0002] repository latest commit in HEAD branch is 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' 
DEBU[0002] retrieving tags for commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' 
DEBU[0002] stepping by commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' 
DEBU[0002] commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' has '0' tags: '[]' 
DEBU[0002] commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' has no valid version tags so it's added to the release scope 
DEBU[0002] trying to infer the identifier to bump based on the commit message of commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' 
DEBU[0002] evaluating commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' against message convention 'conventionalCommitsCustom' 
DEBU[0002] commit message convention 'conventionalCommitsCustom' matches commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' 
DEBU[0002] matching commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' ('refactor(build): nyx configuration
') against bump expression 'major' ('(?s)(?m)^[a-zA-Z0-9_]+(!|.*^(BREAKING( |-)CHANGE: )).*') of message convention 'conventionalCommitsCustom' 
DEBU[0002] bump expression 'major' of message convention 'conventionalCommitsCustom' doesn't match commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' 
DEBU[0002] matching commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' ('refactor(build): nyx configuration
') against bump expression 'minor' ('(?s)(?m)^feat(?!!|.*^(BREAKING( |-)CHANGE: )).*') of message convention 'conventionalCommitsCustom' 
DEBU[0002] bump expression 'minor' of message convention 'conventionalCommitsCustom' doesn't match commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' 
DEBU[0002] matching commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf' ('refactor(build): nyx configuration
') against bump expression 'patch' ('(?s)(?m)(^fix|^refactor)(?!!|.*^(BREAKING( |-)CHANGE: )).*') of message convention 'conventionalCommitsCustom' 
DEBU[0002] bump expression 'patch' of message convention 'conventionalCommitsCustom' matches commit 'cbcf398d0dbf9f8abcfeb03bf19a9b2c8e272dbf', meaning that the 'patch' identifier has to be bumped, according to this commit 
DEBU[0002] retrieving tags for commit '206c2882695f9d8a9c996c6f6efda576d7998a02' 
DEBU[0002] stepping by commit '206c2882695f9d8a9c996c6f6efda576d7998a02' 
DEBU[0002] commit '206c2882695f9d8a9c996c6f6efda576d7998a02' has '0' tags: '[]' 
DEBU[0002] commit '206c2882695f9d8a9c996c6f6efda576d7998a02' has no valid version tags so it's added to the release scope 
DEBU[0002] trying to infer the identifier to bump based on the commit message of commit '206c2882695f9d8a9c996c6f6efda576d7998a02' 
DEBU[0002] evaluating commit '206c2882695f9d8a9c996c6f6efda576d7998a02' against message convention 'conventionalCommitsCustom' 
DEBU[0002] commit message convention 'conventionalCommitsCustom' doesn't match commit '206c2882695f9d8a9c996c6f6efda576d7998a02', skipping 
DEBU[0002] retrieving tags for commit 'f570c08bed0551fcb49ab04583c134a3e464fa99' 
DEBU[0002] stepping by commit 'f570c08bed0551fcb49ab04583c134a3e464fa99' 
DEBU[0002] commit 'f570c08bed0551fcb49ab04583c134a3e464fa99' has '0' tags: '[]' 
DEBU[0002] commit 'f570c08bed0551fcb49ab04583c134a3e464fa99' has no valid version tags so it's added to the release scope 
DEBU[0002] trying to infer the identifier to bump based on the commit message of commit 'f570c08bed0551fcb49ab04583c134a3e464fa99' 
DEBU[0002] evaluating commit 'f570c08bed0551fcb49ab04583c134a3e464fa99' against message convention 'conventionalCommitsCustom' 
DEBU[0002] commit message convention 'conventionalCommitsCustom' doesn't match commit 'f570c08bed0551fcb49ab04583c134a3e464fa99', skipping 
DEBU[0002] retrieving tags for commit '58f52cf8f1552f7ac65cdef9171b9696d735edcd' 
DEBU[0002] stepping by commit '58f52cf8f1552f7ac65cdef9171b9696d735edcd' 
DEBU[0002] commit '58f52cf8f1552f7ac65cdef9171b9696d735edcd' has '0' tags: '[]' 
DEBU[0002] commit '58f52cf8f1552f7ac65cdef9171b9696d735edcd' has no valid version tags so it's added to the release scope 
DEBU[0002] trying to infer the identifier to bump based on the commit message of commit '58f52cf8f1552f7ac65cdef9171b9696d735edcd' 
DEBU[0002] evaluating commit '58f52cf8f1552f7ac65cdef9171b9696d735edcd' against message convention 'conventionalCommitsCustom' 
DEBU[0002] commit message convention 'conventionalCommitsCustom' doesn't match commit '58f52cf8f1552f7ac65cdef9171b9696d735edcd', skipping 
DEBU[0002] retrieving tags for commit 'd1da71923a813f7cf3c63eda9b5ae59bc326cbba' 
DEBU[0002] stepping by commit 'd1da71923a813f7cf3c63eda9b5ae59bc326cbba' 
DEBU[0002] commit 'd1da71923a813f7cf3c63eda9b5ae59bc326cbba' has '1' tags: '[v0.3.28]' 
DEBU[0002] evaluating tag 'v0.3.28': tag is a valid version according to the 'SEMVER' scheme and will be passed to the next evaluation steps. The tag is applied to commit 'd1da71923a813f7cf3c63eda9b5ae59bc326cbba' 
DEBU[0002] evaluating tag 'v0.3.28': tag is greater than previously selected previousVersion tag '<none>' and will be passed to the next evaluation steps. The tag is applied to commit 'd1da71923a813f7cf3c63eda9b5ae59bc326cbba' 
DEBU[0002] evaluating tag 'v0.3.28': the selected release type specifies an additional filter '^(v)?([0-9]\d*)\.([0-9]\d*)\.([0-9]\d*)$' (after template evaluation) for tags and the tag successfully matches so the tag will be used as the previousVersion and the commit 'd1da71923a813f7cf3c63eda9b5ae59bc326cbba' is used as the previousVersionCommit 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa22d97]

goroutine 1 [running]:
github.com/mooltiverse/nyx/modules/go/nyx/command.(*Infer).scanRepository.func1({{{{0xc0013b87b0, 0x24}, {0xc0003f6da1, 0xf}}, {0x1878f421b58, 0xc0003f68f8}}, {{{0xc0013b87e0, 0x24}, {0xc0003f6db0, 0xf}}, ...}, ...})
	/home/runner/work/nyx/nyx/modules/go/nyx/command/infer.go:289 +0x1537
github.com/mooltiverse/nyx/modules/go/nyx/git.goGitRepository.WalkHistory({{0xc00002a094?, 0xc0000ed7f8?}, 0xc0001a3e00?}, 0x0, 0x0, 0xc002029c70)
	/home/runner/work/nyx/nyx/modules/go/nyx/git/go_git_repository.go:1141 +0x89f
github.com/mooltiverse/nyx/modules/go/nyx/command.(*Infer).scanRepository(0xc0001c4dd0, 0xc000022100, 0x0, 0xc000026173, 0xc0001c4330, 0x0, 0xc003103010, 0xc000328090, {0x11a06b0, 0x0, ...}, ...)
	/home/runner/work/nyx/nyx/modules/go/nyx/command/infer.go:227 +0x3d0
github.com/mooltiverse/nyx/modules/go/nyx/command.(*Infer).Run(0xc0001c4dd0)
	/home/runner/work/nyx/nyx/modules/go/nyx/command/infer.go:1463 +0x3ba
github.com/mooltiverse/nyx/modules/go/nyx.(*Nyx).runCommand(0x116a140?, {0xc2dd56, 0x5}, 0x1)
	/home/runner/work/nyx/nyx/modules/go/nyx/nyx.go:328 +0x442
github.com/mooltiverse/nyx/modules/go/nyx.(*Nyx).Infer(0x116a140?)
	/home/runner/work/nyx/nyx/modules/go/nyx/nyx.go:499 +0x65
github.com/mooltiverse/nyx/modules/go/nyx.(*Nyx).Run(0xc000022050?, {0xc2dd56, 0x5})
	/home/runner/work/nyx/nyx/modules/go/nyx/nyx.go:449 +0x1fb
main.main()
	/home/runner/work/nyx/nyx/modules/go/nyx/main/main.go:120 +0x312
```

gradle output (correct):
```
[gregory@opti7080-1 ledger]$ gradle
Project version: v0.3.29

> Task :help

Welcome to Gradle 8.1.

To run a build, run gradle <task> ...

To see a list of available tasks, run gradle tasks

To see more detail about a task, run gradle help --task <task>

To see a list of command-line options, run gradle --help

For more detail on using Gradle, see https://docs.gradle.org/8.1/userguide/command_line_interface.html

For troubleshooting, visit https://help.gradle.org

BUILD SUCCESSFUL in 505ms
1 actionable task: 1 executed
```

**Comment by flyingfishflash** [comment]
In case anyone else comes across this issue, it was solved by adding _collapseVersions: false_ to the mainline release type configuration block

**Comment by flelli** [comment]
Thanks @flyingfishflash ,

I'm glad you found a workaround.
I'll fix this soon.

**Comment by flyingfishflash** [comment]
> Thanks @flyingfishflash ,
> 
> I'm glad you found a workaround. I'll fix this soon.

No problem, great tool you've made!

Might be worth mentioning in the docs that the release publishing of type GITHUB is compatible with gitea/forgejo. At least it works fine for me.

**Comment by flelli** [comment]
Fix available in release [2.3.2](https://github.com/mooltiverse/nyx/releases/tag/2.3.2)
